kubetest/main.go

```diff
@@ -405,7 +416,18 @@ func prepareGcp(kubernetesProvider string) error {
 	// Ensure project is set
 	p := os.Getenv("PROJECT")
 	if p == "" {
-		return fmt.Errorf("KUBERNETES_PROVIDER=%s requires setting PROJECT", kubernetesProvider)
+		p, err := boskos.Acquire("project", "free", "busy")
+		if err != nil {
+			return fmt.Errorf("KUBERNETES_PROVIDER=%s boskos fail to acquire PROJECT:%s", kubernetesProvider, err)
+		}
+
+		go func(c *client.Client, proj string) {
+			for range time.Tick(time.Minute * 5) {
+				if err := c.UpdateOne(p, "busy"); err != nil {
```

Comment:
UpdateOne will check if the updated resource is still hold by the client - https://github.com/kubernetes/test-infra/blob/master/boskos/client/client.go#L115-L122, so if it's called after ReleaseAll() it should return a no resource error, which is probably still fine.